utensor_cgen/transformer/cmsis_nn.py

```diff
@@ -0,0 +1,411 @@
+# -*- coding:utf8 -*-
+r"""CMSIS-NN Transformer
+
+Node fusion and replacement for CMSIS-NN
+
+"""
+import re
+from collections import defaultdict
+from copy import deepcopy
+
+import tensorflow as tf
+import numpy as np
+
+from utensor_cgen.ir import OperationInfo, uTensorGraph
+from utensor_cgen.utils import parse_tensor_name
+
+from .base import Transformer
+
+__all__ = ["CMSIS_NN_Transformer"]
+
+def get_ops_io_info(op_type):
+#please refer to OperatorFactory() in operators.py
+  ops_io_table = dict()
+  ops_io_table["Add"] =                 [[0, 0], [0]]
+  ops_io_table["ArgMax"] =              [[0, 1], [0]]
+  ops_io_table["Dequantize"] =          [[0, 1, 2], [0]]
+  ops_io_table["Max"] =                 [[0, 1], [0]]
+  ops_io_table["QuantizedMaxPool"] =    [[0, 1, 2], [0, 1, 2]]
+  ops_io_table["Min"] =                 [[0, 1], [0]]
+  ops_io_table["QuantizeV2"] =          [[0, 1, 2], [0, 1, 2]]
+  ops_io_table["QuantizedMatMul"] =     [[0, 1, 2, 3, 4, 5], [0, 1, 2]]
+  ops_io_table["MatMul"] =              [[0, 1], [0]]
+  ops_io_table["QuantizedRelu"] =       [[0, 1, 2], [0, 1, 2]]
+  ops_io_table["QuantizedAdd"] =        [[0, 0, 1, 2, 4, 5], [0, 1, 2]]
+  ops_io_table["RequantizationRange"] = [[0, 1, 2], [0, 1]]
+  ops_io_table["Requantize"] =          [[0, 1, 2, 3, 4], [0, 1, 2]]
+  ops_io_table["Reshape"] =             [[0, 1], [0]]
+  ops_io_table["QuantizedReshape"] =    [[0, 1, 2, 3], [0, 1, 2]]
+  ops_io_table["QuantizedConv2D"] =     [[0, 1, 2, 3, 4, 5], [0, 1, 2]]
+  ops_io_table["Const"] =               [None, [0]]
+  ops_io_table["Placeholder"] =         [None, [0]]
+  ops_io_table["Inline"] =              [None, [0]]
+
+  return ops_io_table[op_type]
+
+
+def get_input_nodes(graph, node_name):
```

Comment:
`get_input_nodes` and `get_output_nodes` can be replaced by `graph.ops_info[node].input_nodes` and `graph.ops_info[node].output_nodes` respectively.

Reply:
Oh, it should be `[node.name for node in graph.ops_info[node].input_nodes]`

Reply:
Maybe you can take a look over here:
https://github.com/uTensor/utensor_cgen/commit/e236e0c19be07ad06b592489198b91a1a5d40f8d#diff-100fce0a40bfdc48559207bee1e19641R111
and here
https://github.com/uTensor/utensor_cgen/commit/e236e0c19be07ad06b592489198b91a1a5d40f8d#diff-100fce0a40bfdc48559207bee1e19641R306

Now every `OperationInfo` has a attribute `ugraph` which reference to the graph it belongs to.